packages/better-auth/src/integrations/svelte-kit.ts

```diff
@@ -19,10 +21,13 @@ export const svelteKitHandler = async ({
 	event: { request: Request; url: URL };
 	resolve: (event: any) => any;
 }) => {
-	//@ts-expect-error
-	const { building } = await import("$app/environment")
-		.catch((e) => {})
-		.then((m) => m || {});
+	if (typeof building === "undefined") {
+		//@ts-expect-error
+		const envModule = await import("$app/environment")
+			.catch((e) => {})
+			.then((m) => m || {});
+		building = resolve.building;
```

Comment:
@ping-maxwell what is thought here? resolve is a callback, resolve.building will always be undefined and then this loops as `if (typeof building === "undefined")` is alway true.